lms/djangoapps/grades/grade_utils.py

```diff
@@ -0,0 +1,18 @@
+"""
+This module contains utility functions for grading.
+"""
+from datetime import timedelta
+from django.utils import timezone
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from .config.waffle import ENFORCE_FREEZE_GRADE_AFTER_COURSE_END, waffle_flags
+
+
+def are_grades_frozen(course_key):
+    """ Returns whether grades are frozen for the given course. """
+    if waffle_flags()[ENFORCE_FREEZE_GRADE_AFTER_COURSE_END].is_enabled(course_key):
```

Comment:
only one nit, that if it is possible we should consider early exit if the `if` condition is `False`, Otherwise looks good to me.